requirements-ci.txt

```diff
@@ -14,3 +14,4 @@ requests
 rich
 diastatic-malt
 pandas
+xdsl # used for the Python compiler
```

Comment:
If you mark the tests with ``pytest.mark.external`, you can remove this change and instead, specify xdsl to be installed in the tests workflow file [here](https://github.com/PennyLaneAI/pennylane/blob/9409635483f87937e8b9c343640a612158dad9c6/.github/workflows/interface-unit-tests.yml#L495).
```suggestion
```